22-decay-tree-fitter.md

```diff
@@ -84,9 +84,36 @@ tv__tree->Draw("Dstar_ConsD_D0_M[0]>>h(100,1800,1900)","","", 128, 0);
 
 As expected, the D0 candidates are forced onto their PDG mass value.
 
-The solution to this exercise `ntuple_DTF1.py`, is [available 
-here](./code/22-decay-tree-fitter/ntuple_DTF1.py).
 
 > ## Explore {.challenge}
 > * Look at the `status` variable to check if the fits converged.
-> * Look at the chi2 distribution of the fit 
\ No newline at end of file
+> * Look at the chi2 distribution of the fit 
+
+`DecayTreeFitter` can be told to change some of the hypotheses in the decay tree. This is very useful if you want to slightly change, which decays you want to look at. As an example let's say we want to examine the cabbibo-suppressed decay of the D0 into pi- pi+ instead of K- pi+. For this we add a second fitter, giving it a new name `ConsDpipi`:
+```python
+dtt.Dstar.addTupleTool(TupleToolDecayTreeFitter('ConsDpipi'))
```

Comment:
With `addTupleTool` you shouldn't pass an _instance_ of the tool, instead you should pass a string, e.g.

```python
dtt.Dstar.addTupleTool('TupleToolDecayTreeFitter/ConsDpipi')
```

This is quite nice because the students learnt “instantiate a tool with a string, named with the stuff after the slash” in the interactive DST lesson, when using the tool service.

So, we should remove this line as well:

```python
from Configurables import TupleToolDecayTreeFitter,TupleToolDecay
```

`addBranches` adds a `TupleToolDecay` instance to each branch so that you don't need to, and `addTupleTool` will do the instantiation of TupleTools for you.